src/include/kdb.h.in

```diff
@@ -224,6 +224,9 @@ KeySet *ksVNew(size_t alloc, va_list ap);
 KeySet *ksDup(const KeySet * source);
 int ksCopy(KeySet *dest, const KeySet *source);
 
+uint16_t ksGetRefCount (KeySet * ks);
```

Comment:
There may be cases, where you want to know if you hold an exclusive reference or not. For example, when you want to change the contents of the KeySet without affecting other references.

The `uint16_t` thing can be solved by returning `size_t` instead and mentioning that the actual max number of references may be lower.

Reply:
> What do you think about using more fixed width types in the API in general?

Yeah, we could do that as well.

> and probably some other methods

Yes, we should definitely use fixed size C99 types as much as possible in all of the API. I think this will also make the Rust implementation much easier, as Rust uses fixed sized types by default AFAIK.

Reply:
> There may be cases, where you want to know if you hold an exclusive reference or not. For example, when you want to change the contents of the KeySet without affecting other references.

Then we can simply add a boolean ksIsUnique function. But also this can easily be added in `1.*`.